### ISSUE TITLE
Latest gcc update breaks building Emacs with native-compilation

### ISSUE THREAD

**Issue by dr-scsi** [Homebrew/homebrew-core#195737]
### `brew doctor` output

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Putting non-prefixed coreutils in your path can cause GMP builds to fail.
```

### Verification

- [X] My "`brew doctor` output" above says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` twice and am still able to reproduce my issue.
- [X] This issue's title and/or description do not reference a single formula e.g. `brew install wget`. If they do, open an issue at https://github.com/Homebrew/homebrew-core/issues/new/choose instead.

### `brew config` output

```shell
HOMEBREW_VERSION: 4.4.2
ORIGIN: https://github.com/Homebrew/brew
HEAD: 35ebf4a047b00b2b3de33ba0081d71293cdadede
Last commit: 6 days ago
Core tap JSON: 27 Oct 08:38 UTC
Core cask tap JSON: 27 Oct 08:38 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.3.5 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.5/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 16.0.0 build 1600
Git: 2.47.0 => /opt/homebrew/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.0.1-arm64
CLT: 16.0.0.0.1.1724870825
Xcode: N/A
Rosetta 2: false
```

### What were you trying to do (and why)?

I build Emacs myself with a small script and it seems that updating gcc to 14.2.0_1 breaks the option native-compilation which relies on libgccjit.  It used to work without any problems with 14.2.0 build.

Some info:

```
$ brew info gcc
==> gcc: stable 14.2.0 (bottled), HEAD
GNU compiler collection
https://gcc.gnu.org/
Installed
/opt/homebrew/Cellar/gcc/14.2.0_1 (1,914 files, 459.9MB) *
  Poured from bottle using the formulae.brew.sh API on 2024-10-24 at 21:36:05
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/g/gcc.rb
License: GPL-3.0-or-later WITH GCC-exception-3.1
==> Dependencies
Build: make ✔
Required: gmp ✔, isl ✔, libmpc ✔, mpfr ✔, zstd ✔
==> Options
--HEAD
	Install HEAD version
==> Analytics
install: 130,442 (30 days), 426,448 (90 days), 1,541,340 (365 days)
install-on-request: 67,149 (30 days), 203,183 (90 days), 648,004 (365 days)
build-error: 1,008 (30 days)
```

```
$ brew info libgccjit
==> libgccjit: stable 14.2.0 (bottled), HEAD
JIT library for the GNU compiler collection
https://gcc.gnu.org/
Installed
/opt/homebrew/Cellar/libgccjit/14.2.0 (15 files, 41.0MB) *
  Poured from bottle using the formulae.brew.sh API on 2024-08-26 at 08:59:54
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/lib/libgccjit.rb
License: GPL-3.0-or-later WITH GCC-exception-3.1
==> Dependencies
Required: gmp ✔, isl ✔, libmpc ✔, mpfr ✔, zstd ✔
==> Options
--HEAD
	Install HEAD version
==> Analytics
install: 817 (30 days), 3,358 (90 days), 10,267 (365 days)
install-on-request: 556 (30 days), 2,291 (90 days), 7,072 (365 days)
build-error: 9 (30 days)
```

```
$ gcc --version
Apple clang version 16.0.0 (clang-1600.0.26.3)
Target: arm64-apple-darwin24.0.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

### What happened (include all command output)?

I use a small script to build Emacs which looks like this:

```
git clean -fdx --exclude=ChangeLog

./autogen.sh

GCC_INC=$(realpath $(brew --prefix gcc))
GCC_INC_DIR=${GCC_INC}/include
GCC_LIB_DIR=${GCC_INC}/lib/gcc/current

GCCJIT_INC=$(realpath $(brew --prefix libgccjit))
GCCJIT_INC_DIR=${GCCJIT_INC}/include
GCCJIT_LIB_DIR=${GCCJIT_INC}/lib/gcc/current

GMP_INC=$(realpath $(brew --prefix gmp))
GMP_INC_DIR=${GMP_INC}/include
GMP_LIB_DIR=${GMP_INC}/lib

XML_INC=$(realpath $(brew --prefix libxml2))
XML_INC_DIR=${XML_INC}/include
XML_LIB_DIR=${XML_INC}/lib

CPPFLAGS="-I${GCC_INC_DIR} -I${GCCJIT_INC_DIR} -I${GMP_INC_DIR} -I${XML_INC_DIR} " \
LDFLAGS="-L${GCC_LIB_DIR} -L${GMP_LIB_DIR} -L${XML_LIB_DIR} " \
CFLAGS="-O2 -g0 -pipe" \
./configure \
    --with-ns \
    --without-pop \
    --without-mailutils \
    --with-threads \
    --with-modules \
    --with-native-compilation \
    --without-compress-install

# make -j8
```

After updating to 14.2.0_1, running `configure` for Emacs as above says:

```
checking for gcc_jit_context_acquire in -lgccjit... yes
checking for libgccjit.h... yes
configure: error: The installed libgccjit failed to compile and run a test program using
the libgccjit library; see config.log for the details of the failure.
The test program can be found here:
<https://gcc.gnu.org/onlinedocs/jit/intro/tutorial01.html>.
You can try compiling it yourself to investigate the issues.
Please report the issue to your distribution if libgccjit was installed
through that.
You can find the instructions on how to compile and install libgccjit from
source on this site:
<https://gcc.gnu.org/wiki/JIT>.
```

This is the part from `config.log` (line-breaks added for better legibility):

```
configure:28207: checking for libgccjit.h
configure:28207: gcc -c -g3 -O2 
-isystem /opt/homebrew/Cellar/librsvg/2.58.4/include/librsvg-2.0 
-isystem /opt/homebrew/Cellar/gdk-pixbuf/2.42.12/include/gdk-pixbuf-2.0 
-isystem /opt/homebrew/Cellar/libtiff/4.7.0/include 
-isystem /opt/homebrew/opt/zstd/include 
-isystem /opt/homebrew/Cellar/xz/5.6.3/include 
-isystem /opt/homebrew/Cellar/jpeg-turbo/3.0.4/include 
-isystem /opt/homebrew/Cellar/glib/2.82.2/include 
-isystem /opt/homebrew/Cellar/glib/2.82.2/include/glib-2.0 
-isystem /opt/homebrew/Cellar/glib/2.82.2/lib/glib-2.0/include 
-isystem /opt/homebrew/opt/gettext/include 
-isystem /opt/homebrew/Cellar/pcre2/10.44/include 
-isystem /opt/homebrew/Cellar/cairo/1.18.2/include/cairo 
-isystem /opt/homebrew/Cellar/fontconfig/2.15.0/include 
-isystem /opt/homebrew/opt/freetype/include/freetype2 
-isystem /opt/homebrew/opt/libpng/include/libpng16 
-isystem /opt/homebrew/Cellar/libxext/1.3.6/include 
-isystem /opt/homebrew/Cellar/libxrender/0.9.11/include 
-isystem /opt/homebrew/Cellar/libx11/1.8.10/include 
-isystem /opt/homebrew/Cellar/libxcb/1.17.0/include 
-isystem /opt/homebrew/Cellar/libxau/1.0.11/include 
-isystem /opt/homebrew/Cellar/libxdmcp/1.1.5/include 
-isystem /opt/homebrew/Cellar/pixman/0.42.2/include/pixman-1 
-isystem /opt/homebrew/Cellar/xorgproto/2024.1/include 
-isystem /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/ffi    conftest.c >&5
conftest.c:155:10: fatal error: 'libgccjit.h' file not found
  155 | #include <libgccjit.h>
      |          ^~~~~~~~~~~~~
1 error generated.
configure:28207: $? = 1
```

### What did you expect to happen?

Emacs used to build fine with the way I built it before 14.2.0_1. I expect it to build with native-compilation again.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
Update brew installation and run the bash script above in a local Emacs git repo.
```

**Comment by dr-scsi** [comment]
> It does so: moving to homebrew/core

@MikeMcQuaid - Thanks, and my apologies for submitting to the wrong tracker.

**Comment by Bo98** [comment]
I copy pasted your script and it works fine for me.

The `config.log` you've provided does not seem to match the script you've provided. Where do all those `-isystem` flags come from? Where's all the CPPFLAGS gone?

Here's what I get:

```
gcc -c -O2 -g0 -pipe -I/opt/homebrew/Cellar/libgccjit/14.2.0/include   -I/opt/homebrew/Cellar/gcc/14.2.0_1/include -I/opt/homebrew/Cellar/libgccjit/14.2.0/include -I/opt/homebrew/Cellar/gmp/6.3.0/include -I/include  conftest.c
```

Though with that said, libgccjit usually doesn't need to be supplied via CPPFLAGS: 

https://github.com/emacs-mirror/emacs/blob/9e40d3f2a1c2b5388a4eab72dbe506a21816f69b/configure.ac#L5288-L5296

(though if you happen to have multiple copies of brew installed then this assumes the correct one is first on the PATH)

**Comment by dr-scsi** [comment]
@Bo98 - Boy, that's embarrassing, something went wrong during trying different things, sorry for that.  I don't have multiple copies of brew installed, so with your comment above, the following should just work, right?

```
git clean -fdx --exclude=ChangeLog && ./autogen.sh && ./configure --with-native-compilation
```

Does this work for you?  For me, it says:
```
checking for gcc_jit_context_acquire in -lgccjit... yes
checking for libgccjit.h... yes
configure: error: The installed libgccjit failed to compile and run a test program using
the libgccjit library; see config.log for the details of the failure.
The test program can be found here:
<https://gcc.gnu.org/onlinedocs/jit/intro/tutorial01.html>.
You can try compiling it yourself to investigate the issues.
Please report the issue to your distribution if libgccjit was installed
through that.
You can find the instructions on how to compile and install libgccjit from
source on this site:
<https://gcc.gnu.org/wiki/JIT>.
```

**Comment by carlocab** [comment]
Seems likely to be peculiar to your system and/or build environment, I'm afraid. If the GCC update broke things I'd expect to see issues in places like https://github.com/d12frosted/homebrew-emacs-plus.

**Comment by Bo98** [comment]
> Does this work for you? For me, it says:

Works fine for me.

The weird bit about your log is it does find the header normally:

> ```
> checking for libgccjit.h... yes
> ```

and then fails to find the header immediately afterwards. I'm not sure what's going on there for you.

**Comment by dr-scsi** [comment]
@carlocab - Thanks for your response.  This is what I see:

```
$ brew test --verbose libgccjit
/opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.5/bin/bundle clean
==> Testing libgccjit
==> /opt/homebrew/opt/gcc/bin/gcc-14 -I/opt/homebrew/Cellar/libgccjit/14.2.0/include test-libgccjit.c -o test -L/opt/homebrew/lib/gcc/current -lgccjit
==> ./test
ld: library 'emutls_w' not found
libgccjit.so: error: error invoking gcc driver
NULL result
Error: libgccjit: failed
An exception occurred within a child process:
  Minitest::Assertion: Expected: 0
  Actual: 1
/opt/homebrew/Library/Homebrew/vendor/bundle/ruby/3.3.0/gems/minitest-5.25.1/lib/minitest/assertions.rb:176:in `assert'
/opt/homebrew/Library/Homebrew/vendor/bundle/ruby/3.3.0/gems/minitest-5.25.1/lib/minitest/assertions.rb:216:in `assert_equal'
/opt/homebrew/Library/Homebrew/formula_assertions.rb:31:in `shell_output'
/opt/homebrew/Library/Homebrew/vendor/bundle/ruby/3.3.0/gems/sorbet-runtime-0.5.11608/lib/types/private/methods/call_validation.rb:270:in `bind_call'
/opt/homebrew/Library/Homebrew/vendor/bundle/ruby/3.3.0/gems/sorbet-runtime-0.5.11608/lib/types/private/methods/call_validation.rb:270:in `validate_call'
/opt/homebrew/Library/Homebrew/vendor/bundle/ruby/3.3.0/gems/sorbet-runtime-0.5.11608/lib/types/private/methods/_methods.rb:277:in `block in _on_method_added'
/opt/homebrew/opt/libgccjit/.brew/libgccjit.rb:179:in `block in <class:Libgccjit>'
/opt/homebrew/Library/Homebrew/formula.rb:2813:in `block (3 levels) in run_test'
/opt/homebrew/Library/Homebrew/extend/kernel.rb:539:in `with_env'
/opt/homebrew/Library/Homebrew/formula.rb:2812:in `block (2 levels) in run_test'
/opt/homebrew/Library/Homebrew/formula.rb:1199:in `with_logging'
/opt/homebrew/Library/Homebrew/formula.rb:2811:in `block in run_test'
/opt/homebrew/Library/Homebrew/mktemp.rb:90:in `block in run'
/opt/homebrew/Library/Homebrew/mktemp.rb:90:in `chdir'
/opt/homebrew/Library/Homebrew/mktemp.rb:90:in `run'
/opt/homebrew/Library/Homebrew/vendor/bundle/ruby/3.3.0/gems/sorbet-runtime-0.5.11608/lib/types/private/methods/call_validation.rb:270:in `bind_call'
/opt/homebrew/Library/Homebrew/vendor/bundle/ruby/3.3.0/gems/sorbet-runtime-0.5.11608/lib/types/private/methods/call_validation.rb:270:in `validate_call'
/opt/homebrew/Library/Homebrew/vendor/bundle/ruby/3.3.0/gems/sorbet-runtime-0.5.11608/lib/types/private/methods/_methods.rb:277:in `block in _on_method_added'
/opt/homebrew/Library/Homebrew/formula.rb:3137:in `mktemp'
/opt/homebrew/Library/Homebrew/vendor/bundle/ruby/3.3.0/gems/sorbet-runtime-0.5.11608/lib/types/private/methods/call_validation.rb:270:in `bind_call'
/opt/homebrew/Library/Homebrew/vendor/bundle/ruby/3.3.0/gems/sorbet-runtime-0.5.11608/lib/types/private/methods/call_validation.rb:270:in `validate_call'
/opt/homebrew/Library/Homebrew/vendor/bundle/ruby/3.3.0/gems/sorbet-runtime-0.5.11608/lib/types/private/methods/_methods.rb:277:in `block in _on_method_added'
/opt/homebrew/Library/Homebrew/formula.rb:2805:in `run_test'
/opt/homebrew/Library/Homebrew/test.rb:48:in `block in <main>'
/opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.5/lib/ruby/3.3.0/timeout.rb:186:in `block in timeout'
/opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.5/lib/ruby/3.3.0/timeout.rb:41:in `handle_timeout'
/opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.5/lib/ruby/3.3.0/timeout.rb:195:in `timeout'
/opt/homebrew/Library/Homebrew/test.rb:52:in `<main>'
```

**Comment by carlocab** [comment]
Your install is broken for some reason. Try doing `brew update; brew reinstall gcc libgccjit`

**Comment by dr-scsi** [comment]
@carlocab - Thanks, re-installing `gcc` and `libgccjit` fixed the issue.  This is weird, this is the first time for me that upgrading packages went wrong.  Again, many thanks to you and @Bo98 for your time and support.  I'm closing this report.

**Comment by genehack** [comment]
FWIW I am seeing this exact same error, and reinstalling (multiple times) `gcc` and `libgccjit` is not resolving the problem.

**Comment by genehack** [comment]
> FWIW I am seeing this exact same error, and reinstalling (multiple times) `gcc` and `libgccjit` is not resolving the problem.

I went so far as uninstalling Homebrew, re-installing it, and then re-installing gcc and libgccjit, and `brew test libgccjit` is failing with the same error about not finding libemutls_w. 

Intel Mac.

**Comment by dr-scsi** [comment]
@genehack - Sorry to hear that, I was hoping I'm the only one bitten by this.  I'm reopening this issue so the developers see it.

**Comment by malcolmpurvis** [comment]
I am also seeing the same error, `brew test libgccjit` fails with:

    ld: library 'emutls_w' not found
    libgccjit.so: error: error invoking gcc driver

It is only a problem with Intel Macs.

**Comment by dr-scsi** [comment]
@malcolmpurvis - FWIW, I'm running a M2 MacBook Air.

**Comment by jhorology** [comment]
A week ago, I was able to compile emacs-plus on Intel macOS Sequoia without any problems. I think the only change is the addition of bottle for sequoia([Commit 1f02b0c](https://github.com/Homebrew/homebrew-core/commit/1f02b0c2b90c3a9cf981b75e7d4bc662995d0cb4)).

following workaround works for me:
```shell
# uninstall gcc
brew uninstall --ignore-dependencies gcc
# download sonoma bottle
curl -L -H "Authorization: Bearer QQ==" -o gcc--14.2.0_1.sonoma.bottle.tar.gz https://ghcr.io/v2/homebrew/core/gcc/blobs/sha256:901263863f3f4e8b99f9ffb1cccdb00e5fe74fd1013ac79d37af7ce07f600123
# install gcc from bottle fle
brew install -f gcc--14.2.0_1.sonoma.bottle.tar.gz
```

**Comment by genehack** [comment]
> A week ago, I was able to compile emacs-plus on Intel macOS Sequoia without any problems. 

Can you verify that you built this with native-compilation, and that the native-compilation is actually working? 

The error about the inability to find `emutls_w` semes to be a runtime problem, not a compile time one — a test program will appear to build just fine, but it will not run (and an Emacs that was compiled prior to this will throw that error wehn it tries to natively compile some new code). 

> I think the only change is the addition of bottle for sequoia([Commit 1f02b0c](https://github.com/Homebrew/homebrew-core/commit/1f02b0c2b90c3a9cf981b75e7d4bc662995d0cb4)).
> 
> following workaround works for me:
[snip]

After applying this workaround, `brew test libgccjit` still fails -- albeit with a different error that appears to be due to header file version mismatches.

**Comment by carlocab** [comment]
> following workaround works for me:
> 
> ```shell
> # uninstall gcc
> brew uninstall --ignore-dependencies gcc
> # download sonoma bottle
> curl -L -H "Authorization: Bearer QQ==" -o gcc--14.2.0_1.sonoma.bottle.tar.gz https://ghcr.io/v2/homebrew/core/gcc/blobs/sha256:901263863f3f4e8b99f9ffb1cccdb00e5fe74fd1013ac79d37af7ce07f600123
> # install gcc from bottle fle
> brew install -f gcc--14.2.0_1.sonoma.bottle.tar.gz
> ```

To use the Sonoma bottle, I suggest using
```
brew fetch --bottle-tag sonoma gcc
brew install "$(brew --cache --bottle-tag sonoma gcc)"
```

**Comment by carlocab** [comment]
@genehack, @malcolmpurvis, what is the output of your `brew config` and `brew doctor`?

**Comment by genehack** [comment]
> To use the Sonoma bottle, I suggest using
> 
> ```
> brew fetch --bottle-tag sonoma gcc
> brew install "$(brew --cache --bottle-tag sonoma gcc)"
> ```

This gave me the same result as the other method; `brew test libgccjit` fails with this output:

```
>brew test libgccjit                                                                                                                   
==> Testing libgccjit
==> /usr/local/opt/gcc/bin/gcc-14 -I/usr/local/Cellar/libgccjit/14.2.0/include test-libgccjit.c -o test -L/usr/local/lib/gcc/current -lgccjit
Last 15 lines from /Users/genehack/Library/Logs/Homebrew/libgccjit/test.01.gcc-14:
test-libgccjit.c:27:9: warning: incompatible implicit declaration of built-in function 'fprintf' [-Wbuiltin-declaration-mismatch]
   27 |         fprintf (stderr, "NULL ctxt");
      |         ^~~~~~~
test-libgccjit.c:27:9: note: include '<stdio.h>' or provide a declaration of 'fprintf'
test-libgccjit.c:34:9: warning: incompatible implicit declaration of built-in function 'fprintf' [-Wbuiltin-declaration-mismatch]
   34 |         fprintf (stderr, "NULL result");
      |         ^~~~~~~
test-libgccjit.c:34:9: note: include '<stdio.h>' or provide a declaration of 'fprintf'
test-libgccjit.c:40:9: warning: incompatible implicit declaration of built-in function 'fprintf' [-Wbuiltin-declaration-mismatch]
   40 |         fprintf (stderr, "NULL greet");
      |         ^~~~~~~
test-libgccjit.c:40:9: note: include '<stdio.h>' or provide a declaration of 'fprintf'
test-libgccjit.c:44:5: error: implicit declaration of function 'fflush' [-Wimplicit-function-declaration]
   44 |     fflush (stdout);
      |     ^~~~~~
Error: libgccjit: failed
Warning: Removed Sorbet lines from backtrace!
Rerun with `--verbose` to see the original backtrace
An exception occurred within a child process:
  BuildError: Failed executing: /usr/local/opt/gcc/bin/gcc-14 -I/usr/local/Cellar/libgccjit/14.2.0/include test-libgccjit.c -o test -L/usr/local/lib/gcc/current -lgccjit```

**Comment by genehack** [comment]
> @genehack, @malcolmpurvis, what is the output of your `brew config` and `brew doctor`?

```
> brew doctor
Your system is ready to brew.

> brew config
HOMEBREW_VERSION: 4.4.2-71-g91f3671
ORIGIN: https://github.com/Homebrew/brew
HEAD: 91f3671b16fe62121364c465c14e0dfd088d7ead
Last commit: 25 hours ago
Core tap JSON: 28 Oct 17:36 UTC
Core cask tap JSON: 28 Oct 17:36 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_BAT_THEME: Solarized (dark)
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: e
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.5 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.5/bin/ruby
CPU: octa-core 64-bit icelake
Clang: 16.0.0 build 1600
Git: 2.39.5 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.0.1-x86_64
CLT: 16.0.0.0.1.1724870825
Xcode: 16.0
```

**Comment by carlocab** [comment]
I think this is a mismatch between the SDKs used by `gcc` and `libgccjit`. @Bo98 dispatched a build of `libgccjit` for Sequoia at https://github.com/Homebrew/homebrew-core/actions/runs/11559810507.

Once the PR that produces is merged, try doing
```
brew update && brew reinstall libgccjit
```

**Comment by Bo98** [comment]
Yes, this is my fault as I forgot to build a Sequoia bottle for libgccjit.

Note that using older OS bottles on GCC is strongly not recommended and may lead to compile errors for some projects (including the `brew test libgccjit` test case), so don't use the macOS 14 bottle on macOS 15 - wait a couple hours while I fix this.

**Comment by carlocab** [comment]
Looks like `libgccjit` fails to build because it doesn't expect `libgcc_s.1` to be missing: https://github.com/Homebrew/homebrew-core/actions/runs/11559810507/job/32175335281#step:4:5219

**Comment by malcolmpurvis** [comment]
> @genehack, @malcolmpurvis, what is the output of your `brew config` and `brew doctor`?

    $ brew config
    HOMEBREW_VERSION: 4.4.3
    ORIGIN: https://github.com/Homebrew/brew
    HEAD: 91f3671b16fe62121364c465c14e0dfd088d7ead
    Last commit: 27 hours ago
    Core tap JSON: 28 Oct 19:46 UTC
    Core cask tap JSON: 28 Oct 19:46 UTC
    HOMEBREW_PREFIX: /usr/local
    HOMEBREW_CASK_OPTS: []
    HOMEBREW_MAKE_JOBS: 6
    Homebrew Ruby: 3.3.5 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.5/bin/ruby
    CPU: hexa-core 64-bit kabylake
    Clang: 16.0.0 build 1600
    Git: 2.47.0 => /usr/local/bin/git
    Curl: 8.7.1 => /usr/bin/curl
    macOS: 15.0.1-x86_64
    CLT: 16.0.0.0.1.1724870825
    Xcode: 16.0
    $ brew doctor
    Error: wrong number of arguments (given 4, expected 3)
    /usr/local/Homebrew/Library/Homebrew/formula.rb:220:in `initialize'
    /usr/local/Homebrew/Library/Taps/dunn/homebrew-emacs/Homebrew/emacs_formula.rb:7:in `initialize'
    /usr/local/Homebrew/Library/Homebrew/formulary.rb:571:in `new'
    /usr/local/Homebrew/Library/Homebrew/formulary.rb:571:in `get_formula'
    /usr/local/Homebrew/Library/Homebrew/formulary.rb:800:in `get_formula'
    /usr/local/Homebrew/Library/Homebrew/formulary.rb:1016:in `factory'
    /usr/local/Homebrew/Library/Homebrew/formulary.rb:1107:in `from_keg'
    /usr/local/Homebrew/Library/Homebrew/formulary.rb:1058:in `from_rack'
    /usr/local/Homebrew/Library/Homebrew/diagnostic.rb:730:in `block in check_for_unreadable_installed_formula'
    /usr/local/Homebrew/Library/Homebrew/diagnostic.rb:729:in `each'
    /usr/local/Homebrew/Library/Homebrew/diagnostic.rb:729:in `check_for_unreadable_installed_formula'
    /usr/local/Homebrew/Library/Homebrew/cmd/doctor.rb:59:in `block in run'
    /usr/local/Homebrew/Library/Homebrew/cmd/doctor.rb:52:in `each'
    /usr/local/Homebrew/Library/Homebrew/cmd/doctor.rb:52:in `run'
    /usr/local/Homebrew/Library/Homebrew/brew.rb:94:in `<main>'
    If reporting this issue please do so at (not Homebrew/brew or Homebrew/homebrew-core):
    https://github.com/dunn/homebrew-emacs/issues/new`

`brew doctor` doesn't crash on my other Intel Mac, which uses a week old copy og brew (HEAD: 62c1f5b42bb9959414d1e016f96385f48b9b961d)

**Comment by carlocab** [comment]
This should be fixed in 28b3ba013f1ce63a8e186baaffb22e2176a6751d. Pick up the fix with `brew update && brew reinstall libgccjit`.

If you're still having problems, please open a new issue.
